src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -726,6 +726,35 @@ RequestContext<object> requestContext
                     this.editorSession,
                     eventContext);
             }
+
+            // Convert the editor file glob patterns into an array for the Workspace
+            // TODO Is this thread-safe? this is hardly an atomic operation. But intermediate objects
+            // are costly. Given it's a config only change maybe a race condition won't occur?
+            editorSession.Workspace.ExcludeFilesGlob.Clear();
+            if (configChangeParams.Settings.Files != null && configChangeParams.Settings.Files.Exclude != null)
+            {
+                foreach(string pattern in configChangeParams.Settings.Files.Exclude.Keys)
+                {
+                    if (configChangeParams.Settings.Files.Exclude[pattern] && !editorSession.Workspace.ExcludeFilesGlob.Contains(pattern))
```

Comment:
You just cleared `editorSession.Workspace.ExcludeFilesGlob` on line 733 so we can assume it doesn't contain, right?

Reply:
Yeah, that may be hangover of it making it easy to swap the order of the foreach blocks.  I can remove it.

Reply:
Let's remove it. I don't think the `editorSession.Workspace.ExcludeFilesGlob.Clear()` will be removed/moved so it's not a needed check

Reply:
Done.